go/libraries/doltcore/schema/typeinfo/set.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package typeinfo
+
+import (
+	"encoding/gob"
+	"fmt"
+	"strings"
+
+	"github.com/src-d/go-mysql-server/sql"
+
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+const (
+	setTypeParam_Collation = "collate"
+	setTypeParam_Values    = "vals"
+)
+
+// This is a dolt implementation of the MySQL type Set, thus most of the functionality
+// within is directly reliant on the go-mysql-server implementation.
+type setType struct {
+	sqlSetType sql.SetType
+}
+
+var _ TypeInfo = (*setType)(nil)
+
+func CreateSetTypeFromParams(params map[string]string) (TypeInfo, error) {
+	var collation sql.Collation
+	var err error
+	if collationStr, ok := params[setTypeParam_Collation]; ok {
+		collation, err = sql.ParseCollation(nil, &collationStr, false)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, fmt.Errorf(`create set type info is missing param "%v"`, setTypeParam_Collation)
+	}
+	var values []string
+	if valuesStr, ok := params[setTypeParam_Values]; ok {
+		dec := gob.NewDecoder(strings.NewReader(valuesStr))
+		err = dec.Decode(&values)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, fmt.Errorf(`create set type info is missing param "%v"`, setTypeParam_Values)
+	}
+	sqlSetType, err := sql.CreateSetType(values, collation)
+	if err != nil {
+		return nil, err
+	}
+	return &setType{sqlSetType}, nil
+}
+
+// ConvertNomsValueToValue implements TypeInfo interface.
+func (ti *setType) ConvertNomsValueToValue(v types.Value) (interface{}, error) {
+	if val, ok := v.(types.String); ok {
```

Comment:
Not sure about storing these as strings. The MySQL code is going to limit these to 64 elements specifically so we could use a uint64 to store -- difference in storage cost is enormous.

Let's table the discussion for now by sabotaging the type until we can come to a consensus.